src/expr/src/relation/func.rs

```diff
@@ -498,6 +517,8 @@ pub enum AggregateFunc {
     /// layer, this function filters out `Datum::Null`, for consistency with
     /// the other aggregate functions.
     JsonbObjectAgg,
+    /// Accumulates `Datum::Array`s into a single `Datum::Array`.
+    ArrayAgg,
```

Comment:
What do you think about calling this `ArrayConcat` or somesuch to get at the fact that it's different from SQL's `array_agg`?